src/js/controller/model.js

```diff
@@ -186,9 +185,7 @@ const Model = function() {
         if (autoStart !== undefined) {
             this.set('autostart', autoStart);
         }
-
-        const autoStartOnMobile = OS.mobile && this.get('autostart');
-        this.set('playOnViewable', autoStartOnMobile || this.get('autostart') === 'viewable');
```

Comment:
Why is this logic being removed?

"autoPause" should only result in playback resuming when the player becomes visible, if it was auto paused. 

I wouldn't overload "playOnViewable" here. I would actually get rid of it and base all logic on "autostart", "autoPause" and user interaction which disables "playOnViewable" ("autostartViewable"). This logic doesn't belong here in model.js - keep it all in controller.js even if you want to keep the "playOnViewable" property.

Reply:
The logic is replaced by a looser definition of `playOnViewable`. This way, the `_checkPlayOnViewable` handler in `controller.js` is executed for `autostart: true` players as well. This handler then pauses the video if not viewable.

Reply:
This turns on viewable autostart for desktop players when `autostart` is set to `true` - Which changes behavior:

- if autostart is not allowed then view-ability changes
- when autostart is allowed ... ?

And what about this statement in controller `_checkAutoStart`?

```js
            // Autostart immediately if we're not waiting for the player to become viewable first.
            if (_model.get('autostart') === true && !_model.get('playOnViewable')) {
                _autoStart();
            }
```